src/gstring.rs

```diff
@@ -0,0 +1,440 @@
+// Copyright 2018, The Gtk-rs Project Developers.
+// See the COPYRIGHT file at the top-level directory of this distribution.
+// Licensed under the MIT license, see the LICENSE file or <http://opensource.org/licenses/MIT>
+
+use std::cmp::Ordering;
+use std::ffi::{CStr, CString, OsStr};
+use std::fmt;
+use std::ops::Deref;
+use std::os::raw::c_char;
+use std::ptr;
+use std::string::String;
+use translate::*;
+use types::{StaticType, Type};
+use libc;
+
+use gobject_ffi;
+use ffi as glib_ffi;
+use value::{FromValueOptional, Value, SetValueOptional, SetValue};
+
+pub struct GString {
+    s: Option<CString>,
+    ptr: *const c_char,
+    length: usize
+}
+
+impl GString {
+    pub unsafe fn new(ptr: *const c_char) -> Self {
```

Comment:
This should probably be `*mut c_char` as we take ownership of the pointer that is passed in